### DISCUSSION TITLE
مسدود کردن حملات SSH

### DISCUSSION THREAD

**Discussion by hosseiiin20** [hiddify/Hiddify-Manager#3185] [General سایر]
**جلوگیری از حملات برای رمز ssh** 
تست شده روی هیدیفای - ubuntu 20.04
اگر 2 بار رمز اشتباه بزنه طی دو ساعت به مدت 48 ساعت ای پی بن میشه

**سلب مسئولیت**
این یک اسکریپت جانبی می باشد که فایروال iptables آی پی حمله کننده به سرویس ssh را مسدود میکند
مسئولیت نصب و مشکلات احتمالی به عهده خودتون می باشد و این یک پیشنهاد نصب توسط شما نمی باشد

----------------------------------

_روش های کمکی دیگر هم هست مانند 
PermitEmptyPasswords no
new sudo user and disable root login
strong password
ssh key
که خودتون قبلا باید انجام بدید_

----------------------------------

1. نصب fail2ban

apt-get install fail2ban

----------------------------------
2.
   
nano /etc/fail2ban/jail.local

----------------------------------
3. به انتهای فایل رفته و خطوط زیر را اضافه کنید:
   
[sshd]

enabled = true
port = ssh
filter = sshd
logpath = /var/log/auth.log
maxretry = 2
bantime = 172800
findtime = 7200
ignoreip = 127.0.0.1/8 ::1
bannedaction = iptables-xt-wrapper banip %s

----------------------------------

- **توضیحات**

- maxretry: تعداد دفعات ناموفق
- findtime: در یک زمان مشخص به ثانیه
- bantime: مقدار زمان بن شدن به ثانیه
- ignoreip : وایت لیست

----------------------------------
4.

sudo systemctl restart fail2ban
   
----------------------------------
نکته بسیار مهم :

اگر زمانی قصد اعمال تغییرات یا آپدیت پنل کردید قبل از آن بکاپ بگیرید :
زیرا پنل بعد از اعمال تغییرات تنظیمات iptable را تغییر میدهد و باعث مشکل میشود
این کار را تا زمانی که دوستان عزیز هیدیفای این باگ را حل کنند انجام دهید

sudo sh -c "iptables-save > /etc/iptables/backup_v4"

بعد از اینکه پنل بالا اومد فایل بکاپ را ریستور میکنید

sudo iptables-restore < /etc/iptables/backup_v4

----------------------------------

نمایش لاگ fail2ban

sudo nano /var/log/fail2ban.log
   
نمایش لیست بن شدگان در iptable توسط fail2ban

sudo iptables -S | grep f2b

مشاهده حملات

sudo grep 'Failed password' /var/log/auth.log

مشاهده کسانی که موفق به ورود به اس اس اچ شدند

grep "sshd.*Accepted password" /var/log/auth.log

درآوردن یک آی پی از مسدودی با یک مثال

sudo fail2ban-client set sshd unbanip 5.130.72.186

**Comment by hiddify-com** [comment]
ایده خوبیه ولی نمیتونیم تو پنل پیشفرض بذاریم چون اگه اینکار رو بکنیم یه فینگرپرینت اضافه میکنیم برای شناسایی سرور
پیشنهاد اینه از پابلیک کی استفاده کنند یا رمز قوی ولی هر حرکتی بزنم یه فینگرپرینت خواهد بود

**Reply by ERFAN-Freak** [reply]
لطف میکنید اموزش فینگرپرینت بزارید با تشکر از شما
